lib/logger/lib/logger/config.ex

```diff
@@ -1,95 +1,72 @@
 defmodule Logger.Config do
   @moduledoc false
 
-  # TODO Remove this once we support Erlang/OTP 22+ exclusively.
-  @compile {:no_warn_undefined, [:persistent_term, :counters]}
-
   @behaviour :gen_event
   @name __MODULE__
-  @table __MODULE__
-  @data_key __MODULE__
-  @deleted_handlers_key __MODULE__.DeletedHandlers
-  @counter_pos 1
   @update_counter_message {__MODULE__, :update_counter}
 
+  alias Logger.Counter
+
   def configure(options) do
     :gen_event.call(Logger, @name, {:configure, options})
   end
 
+  def default_config do
+    sync_threshold = Application.fetch_env!(:logger, :sync_threshold)
+    discard_threshold = Application.fetch_env!(:logger, :discard_threshold)
+    level = Application.fetch_env!(:logger, :level)
+    sasl_reports? = Application.fetch_env!(:logger, :handle_sasl_reports)
+
+    %{
+      level: level,
+      utc_log: Application.fetch_env!(:logger, :utc_log),
+      truncate: Application.fetch_env!(:logger, :truncate),
+      translators: Application.fetch_env!(:logger, :translators),
+      thresholds: {sync_threshold, discard_threshold},
+      sasl: sasl_reports?
+    }
+  end
+
   def add_translator(translator) do
-    :gen_event.call(Logger, @name, {:add_translator, translator})
+    update_translators(fn t -> [translator | List.delete(t, translator)] end)
```

Comment:
This introduces races conditions if two processes are trying to add or remove translators, only one of them would in. So we have to move it back to Logger.Config to make the operation atomic.